lib/Zonemaster/CLI.pm

```diff
@@ -358,9 +356,9 @@ sub run {
     }
 
     # These two must come after any profile from command line has been loaded
-    # to override the profile setting
-    Zonemaster::Engine::Profile->effective->set( q{net.ipv4}, 0+$self->ipv4 );
-    Zonemaster::Engine::Profile->effective->set( q{net.ipv6}, 0+$self->ipv6 );
+    # to override the profile setting, but only if selected.
+    Zonemaster::Engine::Profile->effective->set( q{net.ipv4}, 0+$self->ipv4 ) if defined ($self->ipv4);
```

Comment:
The `else` statement I see :

```
else {
    Zonemaster::Engine::Profile->effective->set( q{net.ipv4}, 1 );
}
```

Reply:
That "else" should first check if any of ipv4/ipv6 is enabled, if not set both. Otherwise we will, again, override the setting in any selected config file.

I will check and come with an updated version.

Reply:
Oh yes I didn't see that it was as twisted as that.

I think this cumbersome code covers every cases (maybe some perl magic might make it lighter)

```
    if ( defined ($self->ipv4) ) {
        Zonemaster::Engine::Profile->effective->set( q{net.ipv4}, $self->ipv4 );
    } else {
        if ( ! defined ($self->ipv6) ) {
            Zonemaster::Engine::Profile->effective->set( q{net.ipv4}, 1 );
        }
    }

    if ( defined ($self->ipv6) ) {
        Zonemaster::Engine::Profile->effective->set( q{net.ipv6}, $self->ipv6 );
    } else {
        if ( ! defined ($self->ipv4) ) {
            Zonemaster::Engine::Profile->effective->set( q{net.ipv6}, 1 );
        }
    }
```

Reply:
This will also be wrong, because if you have selected a profile where IPv6 has been disabled, then your suggestion will enable IPv6 unless you explicitly have selected --no-ipv6.

I will proved an update.

Reply:
This is really twisted ^^. I'll let you find a solution ;)